content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+already made solutions such as [create-react-library][105].
+
+You will end up with a bundle made of different files.
+
+A typical output might look like this:
+
+```
+/dist
+  - bundle.js
+  - bundle.css
+  images/
+    - dog.jpg
+```
+
+This is what you as an Author would publish to [NPM][102].
+
+`bundle.css` will use the image by doing `url(./images/dog.jpg)`, so
+both files need to be served in the following URLs
+
+- `yourSite.com/public/bundle.css`
+- `yourSite.com/public/images/dog.jpg`
+
+`public` can be any path with a trivial length, the important part is
+that they are both served from [the same path][106]
+
+Aside from properly document how the files inside your package
+relate to each other there is not much else you can do.
+
+You can also use this strategy by allowing access to your module via a CDN,
+by which _Component Consumers_ can simply reference the module.
+
+#### As Component Consumers
+
+This is where this approach has a hard time providing a good Consumer experience.
+
+Consumers need to:
+
+- `import module from 'module'` for Javascript, in every file that uses the module
+- Import the CSS by using either:
+  - A bundler like Webpack:
+    - `import "module.css"` only once in your Application
+  - HTML link:
+    - `cp ./node_modules/module/module.css ./public/images/dog.jpg`: copy the CSS file to your public directory
+    - `<link type="text/css" rel="stylesheet" href="public/module.css">`: load the CSS in your page.
+- Serve images at the path the module's CSS and/or Javascript files expect them, in this case at `./images/dog.jpg`
+  - `cp ./node_modules/module/images/dog.jpg ./public/images/dog.jpg`
+  - Serve `public/images/dog.jpg` by properly configuring your Application
+
+We assume that `public` is your App's public directory from where you serve your
+static files and assets, for convenience the directory is called `public` and the
+URL path is also called `public`.
+
+Notice how to _Consume_ a simple module you need to do a lot of steps and
+while there are tools to automate the process such as [Webpack][2], [Grunt][109], [Gulp][108]
+or a package like [npm-assets][107], the overall experience is not optimal.
+
+An obvious additional problem is potential _name conflicts_,
+for example if two libraries need to serve `images/dog.jpg`.
+
+_Component Authors_ might make the module available via a CDN and that might
+remove the need to fiddle around with CSS and Images but you will need to
+access the library via a Browser Global like the good old days.
+
+### Inline
+
+#### As Component Authors
+
+The bundler setup is about the same as with **Not-inline**.
+You wont be able to use [create-react-library][105] but you can probably
+find other tools that allow you to customize them to Inline everything.
+
+The Webpack configuration is actually not that hard and a working example
+will be provided at the end of this post.
+
+The key Webpack configuration is the inclusion of [url-loader][11] to
```

Comment:
It looks like the last 3 paragraphs start with a variation of "the webpack configuration...". Maybe they can be merged in a way that makes it flow?

Reply:
Simplified a bit more, a reduced to 2 jajajaja